server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -444,6 +451,11 @@ private IndexCommit getStartingCommitPoint() throws IOException {
                     // TODO: Asserts the starting commit is a safe commit once peer-recovery sets global checkpoint.
                     startingIndexCommit = CombinedDeletionPolicy.findSafeCommitPoint(existingCommits, lastSyncedGlobalCheckpoint);
                 }
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Open engine with the safe commit [{}], global checkpoint [{}], existing commits [{}]",
```

Comment:
we will already log the safe commits and deletions in the onInit of the CombinedDeletionPolicy, maybe this is an overkill?